### ISSUE TITLE
How set ffmpeg pad

### ISSUE THREAD

**Issue by MohammadRezaei92** [bytedeco/javacv#1440]
I want to convert video with change video size but keep aspect ratio like this command:
`ffmpeg -i input -vf "scale=1280:720:force_original_aspect_ratio=decrease,pad=1280:720:(ow-iw)/2:(oh-ih)/2" output`
How should I do it?

**Comment by saudet** [comment]
If that command already does exactly what you need, you can execute it this way:
http://bytedeco.org/javacpp-presets/ffmpeg/apidocs/org/bytedeco/ffmpeg/ffmpeg.html

**Comment by MohammadRezaei92** [comment]
> If that command already does exactly what you need, you can execute it this way:
> http://bytedeco.org/javacpp-presets/ffmpeg/apidocs/org/bytedeco/ffmpeg/ffmpeg.html

Is there a way to do that with FrameGrabber and FrameRecorder?

**Comment by saudet** [comment]
With `FFmpegFrameFilter`, yes.

**Comment by MohammadRezaei92** [comment]
> With `FFmpegFrameFilter`, yes.

How?

**Comment by saudet** [comment]
Something like this: https://github.com/bytedeco/javacv/blob/master/samples/DeinterlacedVideoPlayer.java

**Comment by MohammadRezaei92** [comment]
> Something like this: https://github.com/bytedeco/javacv/blob/master/samples/DeinterlacedVideoPlayer.java

I used this method but I always got this exception:
`avfilter_graph_create_filter() error -22: Cannot create video buffer source.`

Here is my code:
```
val filterString = "scale=${grabber.imageWidth}:${grabber.imageHeight}" +
                            ":force_original_aspect_ratio=decrease," +
                            "pad=${grabber.imageWidth}:${grabber.imageHeight}:(ow-iw)/2:(oh-ih)/2"
                    filter = FFmpegFrameFilter(filterString,grabber.imageWidth,grabber.imageHeight)
                    filter.pixelFormat = grabber.pixelFormat
                    filter.start()
                    var frame: Frame?
                    while (grabber.grabFrame().also { frame = it } != null) {
                        recorder.timestamp = grabber.timestamp
                        filter.push(frame)
                        recorder.record(filter.pull())
                    }
                    filter.stop()
                    recorder.stop()
                    grabber.stop()
```

**Comment by saudet** [comment]
Look at the error message displayed on the console for more information about what is wrong.

**Comment by saudet** [comment]
If you don't see anything displayed on the console, make sure `FFmpegLogCallback.set()` gets called.

**Comment by MohammadRezaei92** [comment]
I solved it. 
My problem was my app language was Farsi, which caused numbers to appear in Farsi mode in `FFmpegFrameFilter`.
Solution: I made a copy of the `FFmpegFrameFilter` class and added `Locale.US` to `String.format`s

**Comment by saudet** [comment]
I see, thanks for reporting! Could you send a pull request with the fix?

**Comment by saudet** [comment]
Thanks for the fix! It has been released with JavaCV 1.5.4.
